src/s1reader/s1_orbit.py

```diff
@@ -267,6 +319,11 @@ def get_orbit_file_from_dir(zip_path: str, orbit_dir: str, auto_download: bool =
 
     orbit_file = get_orbit_file_from_list(zip_path, orbit_file_list)
 
+    # if no orbit file in the list, try to find RESORB files and concatenate
+    if not orbit_file and concat_resorb:
+        print('Attempting to concatenate RESORB files in the orbit file list.')
+        orbit_file = concatenate_resorb_from_list(zip_path, orbit_file_list)
+
     if orbit_file:
         return orbit_file
```

Comment:
Just wanna check my understanding on the logic flow here:

1. we use `get_orbit_file_from_list` to check if an existing orbit in the directory covers the time span we need (a full orbit period + 60 seconds before the sensing start, sensing end)
2. then we try to concatenate two RESORBs from the list of existing ones with `concatenate_resorb_from_list`
3. then if that fails we call `retrieve_orbit_file`, which will also eventually concatenate them after downloading?

I think the part that threw me off at first was that we're checking for RESORB coverage in two different ways, once with the list of existing files and `_is_orbitfile_cover_timeframe`, and once with the `get_orbit_dict` checks.

Reply:
There has been a change in the logic flow. based on the SAS tests and the discussions from ADT/SDS meeting
- We are going to provide the RESORB pair as a list. I've kept the orbit file concatenation feature, but it will be turned off by default.
- search for RESORB pair will be attempted in `concatenate_resorb_from_list` and `retrieve_orbit_file` because some R2 SAS uses `concatenate_resorb_from_list()`. Doing the RESORB pair search in `get_orbit_file_from_dir` level will make the SAS not be able to find the RESORB pair.
- In searching for existing orbit file, `get_resorb_pair_from_list` will be used. `get_orbit_dict` will be used for finding the orbit files from internet.